Vvm add display user

aurelienCastellarnau opened the pull request:
- Boom le contexte de test (application console)
- Ajoute un système de routing basé sur un combo button/page (simpliste à souhait...)
- On part du principe qu'on va réaliser les premiers tests manuellement
- On tente de set une TextBox Info avec le contenu des exceptions => ca marche pas, il voit la TextBox comme null à la construction...

Commit:
feature(MainWindow): Dans App/cs file (app.xaml.cs) on récupère la MainWindow. Ca nous permettra probablement d'insérer de la logique globale à l'app avant de lancer la vue (session? middleware? config? rien? on s'en fou c'est chouette d'avoir la MainWindow non?)

Commit:
feature(MainWindow): Navigation. On peut partir sur ca pour générer nos vues. Chaque composant serait une page déclenchée par des actions et remplissant le content. On peut facilement s'inspirer de ce qu'on fait sur angular.

Kumatetsu committed:
ajout d'un projet console pour tester et ajouter des datas facilement

Commit:
boom application console

Commit:
feature(component_behind): Composants derrière les vues. Logique protégée par des try catch settant une variable info... l'idée me parait bonne mais ne fonctionne pas car l'objet Vue Info est null à la construction.. étrange....

Kumatetsu commented on Movienet/ClassLibrary1/MainClass.cs:
```diff
@@ -8,13 +8,10 @@ namespace ClassLibrary1
-            User user = new User();
 
-            ctx.SaveChanges();
+        public static void Main(string[] args)
```
du coup on est en public parce que il fallait que je force l'accès pour que l'autre proj y ai accès. (ça a touché pas mal de functions du coup à cause du lien entre les différentes classes/interfaces/abstract, l'un ne peux pas être public et contenir des function plus restreinte)

Kumatetsu commented on Movienet/Movienet/AddUser.xaml.cs:
```diff
@@ -0,0 +1,53 @@
+            IUserDao uDao = Services.GetUserDao();
+            String info = "Adding a user";
+            ClassLibrary1.User checkUser = new User();
+            ClassLibrary1.User nUser = new User();
```
ton nUser sert pas encore, j'imagine que dans le futur il servira ?

Kumatetsu replied:
ah non mabad, tu l'utilise pour create et recup dans le check user. Pas fan de ça, nUser sur un create devrait pas changer et c'est toujours lui, donc on peut en avoir qu'un seul :P

Kumatetsu commented on Movienet/Movienet/MainWindow.xaml.cs:
```diff
@@ -10,19 +10,31 @@
         }
+
+        // évênements onClick définis dans les buttons
+        private void goToDisplayUser(object sender, RoutedEventArgs e)
```
coooooode behind!

aurelienCastellarnau replied:
ah ouai merde... j'avais zappé cuila... bouge pas, je m'en occupe plus tard ;)

Commit:
delete(ClassLibrary1): suppression du data model dans le processus de dev' pour repasser sur une base neuve.

Commit:
feature(versionning_mdf_file): on versionne le fichier de data.

Commit:
feature(new_model_project): J'ai recréer le projet back pour le faire fonctionner (2x ^^).
Points important: j'ai galérer sur le 1er jet avec l'absence de .mdf, du coup il est versionné dans le précédent commit.
Les propriétés étant un mapping de la base, entity générant des propriétés avec majuscule, les propriétés dans les tables doivent avoir des majuscules aussi...

Commit:
feature(display/add/User): intégration du nouveau back + ajout du pattern locator + vue (pages) pour display users et add user. Binding avec les méthodes de la dao.

aurelienCastellarnau commented:
PTAL

Kumatetsu commented:
Partie Back -> OK bien joué pour l'intégration coté front et la communication des deux !

Partie Front -> Un petit screen de ce que ça donne mais côté technique ça m'a l'air de carrément être du mvvm :P
